python/ray/util/sgd/v2/session.py

```diff
@@ -0,0 +1,116 @@
+import queue
+import time
+import threading
+
+from ray.util.sgd.v2.constants import TIME_THIS_ITER_S
+
+
+class Session:
+    """Holds information for training on each worker."""
+
+    def __init__(self, training_thread: threading.Thread, world_rank: int):
+        # The Thread object that is running the training function.
+        self.training_thread = training_thread
+        self.world_rank = world_rank
+
+        # This lock is used to control the execution of the training thread.
+        self.continue_lock = threading.Semaphore(0)
+
+        # Queue for sending results across threads.
+        self.result_queue = queue.Queue(1)
+
+        self.last_report_time = time.time()
+
+    def report(self, **kwargs):
+        """Adds kwargs to the queue to be consumed by main thread."""
+        current_time = time.time()
+        time_this_iter = current_time - self.last_report_time
```

Comment:
Should we updated `self.last_report_time` here?

Reply:
Yep good catch! Fixed.